components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/AbstractResponseTypeHandler.java

```diff
@@ -126,6 +126,9 @@ && hasValidationByApplicationScopeValidatorsFailed(oauthAuthzMsgCtx)) {
             String[] approvedScopes = oauthAuthzMsgCtx.getApprovedScope();
             String[] requestedOIDCScopes =
                     OAuth2Util.getRequestedOIDCScopes(oauthAuthzMsgCtx.getAuthorizationReqDTO().getScopes());
+            if (log.isDebugEnabled()) {
+                log.debug("Adding OIDC scopes back to approved scopes after engaging global scope validators");
```

Comment:
We should say filtered scopes(The current variable name is a bit misleading) isn't it?
We cannot override the approved scopes set, isn't it?